src/cloud_provider/aws/kubernetes/mod.rs

```diff
@@ -119,1675 +109,1168 @@ pub struct Options {
+                        Err(e) => {
+                            OperationResult::Retry(
+                                EngineError::new_cannot_get_k8s_api_custom_metrics(event_details.clone(), e))
                         }
+                    };
+                });
+
+                match wait_engine_job_finish {
+                    Ok(_) => {
+                        kubernetes.logger().log(EngineEvent::Info(event_details.clone(), EventMessage::new_from_safe("No current running jobs on the Engine, infrastructure pause is allowed to start".to_string())));
+                    }
+                    Err(Operation { error, .. }) => {
+                        return Err(error);
+                    }
+                    Err(retry::Error::Internal(msg)) => {
+                        return Err(EngineError::new_cannot_pause_cluster_tasks_are_running(event_details, Some(CommandError::new_from_safe_message(msg))));
                     }
                 }
-                false => self.logger().log(EngineEvent::Info(event_details.clone(), EventMessage::new_from_safe("Engines are running Client side, but metric history flag is disabled. You will encounter issues during cluster lifecycles if you do not enable metric history".to_string()))),
             }
+            false => kubernetes.logger().log(EngineEvent::Info(event_details.clone(), EventMessage::new_from_safe("Engines are running Client side, but metric history flag is disabled. You will encounter issues during cluster lifecycles if you do not enable metric history".to_string()))),
         }
+    }
 
-        let mut terraform_args_string = vec!["apply".to_string(), "-auto-approve".to_string()];
-        for x in tf_workers_resources {
-            terraform_args_string.push(format!("-target={}", x));
-        }
-        let terraform_args = terraform_args_string.iter().map(|x| &**x).collect();
-
-        self.send_to_customer(
-            format!("Pausing EKS {} cluster deployment with id {}", self.name(), self.id()).as_str(),
-            &listeners_helper,
-        );
-        self.logger().log(EngineEvent::Info(
-            event_details.clone(),
-            EventMessage::new_from_safe("Pausing EKS cluster deployment.".to_string()),
-        ));
+    let mut terraform_args_string = vec!["apply".to_string(), "-auto-approve".to_string()];
+    for x in tf_workers_resources {
+        terraform_args_string.push(format!("-target={}", x));
+    }
+    let terraform_args = terraform_args_string.iter().map(|x| &**x).collect();
+
+    kubernetes.send_to_customer(
+        format!(
+            "Pausing {} {} cluster deployment with id {}",
+            kubernetes.kind(),
+            kubernetes.name(),
+            kubernetes.id()
+        )
+        .as_str(),
+        &listeners_helper,
+    );
+
+    kubernetes.logger().log(EngineEvent::Info(
+        event_details.clone(),
+        EventMessage::new_from_safe("Pausing EKS cluster deployment.".to_string()),
```

Comment:
OPT: Maybe a miss? 

Again, `EventMessage::new_from_safe("Pausing cluster deployment.".to_string())` should be enough

Reply:
ack